src/Extensions/Features/src/FeatureCollectionExtensions.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.AspNetCore.Http.Features;
+
+/// <summary>
+/// Extension methods for getting feature from <see cref="IFeatureCollection"/>
+/// </summary>
+public static class FeatureCollectionExtensions
+{
+    /// <summary>
+    /// Get feature of type <typeparamref name="TFeature"/> from the <see cref="IFeatureCollection"/>
+    /// </summary>
+    /// <param name="featureCollection"></param>
+    /// <returns>Feature object type</returns>
+    /// <exception cref="InvalidOperationException">Thrown when asked for unregistered feature type</exception>
+    public static TFeature GetRequiredFeature<TFeature>(this IFeatureCollection featureCollection)
+        where TFeature : notnull
+    {
+        if (featureCollection is null)
+        {
+            throw new ArgumentNullException(nameof(featureCollection));
+        }
+
+        return featureCollection.Get<TFeature>() ?? throw new InvalidOperationException($"{nameof(TFeature)} is not available");
```

Comment:
```suggestion
        return GetRequiredFeature(featureCollection, typeof(TFeature));
```

Reply:
No we should be using the generic overload. It's optimized.